### PR TITLE
⬆️ Updates and fix bilibili

### PR DESCRIPTION
**Pull request by Kotoki1337**
#1554

image
image
image
image
image

### CODE REVIEW

File: websites/B/bilibili/presence.ts [hunk 1]
```diff
@@ -103,37 +103,50 @@ presence.on("UpdateData", async () => {
...
+      }
       presenceData.details = title.innerText;
-      presenceData.state = "By user: " + user.innerText;
+      presenceData.state = user.innerText;
+
       //播放历史
     } else if (document.location.pathname == "/account/history") {
       presenceData.startTimestamp = browsingStamp;
-      presenceData.details = "Viewing history";
+      presenceData.details = "Their history";
```
Comment (Bas950) [inline websites/B/bilibili/presence.ts:130]: `Viewing their history` might be better
Viewing history or Their history is both not really good.

File: websites/B/bilibili/presence.ts [hunk 2]
```diff
@@ -720,13 +286,12 @@ presence.on("UpdateData", async () => {
         "#app > div.page-container.p-rel > div.top-header.p-rel > div.tag-title-content.fs-28.ls-0 > div.tag-title.d-i-block"
       );
       presenceData.startTimestamp = browsingStamp;
-      presenceData.details = "Browsing dynamic";
+      presenceData.details = "Dynamic";
       presenceData.state = "Tag: " + title.innerText;
     } else {
       presenceData.startTimestamp = browsingStamp;
-      presenceData.details = "Browsing for dynamic";
+      presenceData.details = "Their dynamic";
```
Comment (Bas950) [inline websites/B/bilibili/presence.ts:293]: Their dynamic doesn't sound correct.

File: websites/B/bilibili/presence.ts [hunk 3]
```diff
@@ -219,475 +232,28 @@ presence.on("UpdateData", async () => {
...
-      presenceData.details = "Viewing TV series";
-      presenceData.state = "└ China domestic";
-    } else if (document.location.pathname.includes("/v/tv/overseas/")) {
-      presenceData.startTimestamp = browsingStamp;
-      presenceData.details = "Viewing TV series";
-      presenceData.state = "└ China overseas";
-    } else if (document.location.pathname.includes("/tv/index/")) {
-      presenceData.startTimestamp = browsingStamp;
-      presenceData.details = "Viewing TV series";
-      presenceData.state = "└ index";
```
Comment (Bas950) [inline websites/B/bilibili/presence.ts:690]: Why was this all removed?

File: websites/B/bilibili/presence.ts [hunk 4]
```diff
@@ -764,27 +329,24 @@ presence.on("UpdateData", async () => {
...
+      presenceData.state = user.innerText;
     } else {
       presenceData.startTimestamp = browsingStamp;
       presenceData.details = "Browsing Someone's space";
     }
   } else if (document.location.hostname == "account.bilibili.com") {
     if (document.location.pathname.includes("/")) {
       presenceData.startTimestamp = browsingStamp;
-      presenceData.details = "Viewing account setting";
+      presenceData.details = "Their account settings";
```
Comment (Bas950) [inline websites/B/bilibili/presence.ts:343]: `Viewing their settings` might be better

File: websites/B/bilibili/presence.ts [hunk 5]
```diff
@@ -764,27 +329,24 @@ presence.on("UpdateData", async () => {
...
     }
     //创作中心
   } else if (document.location.hostname == "member.bilibili.com") {
     if (document.location.pathname.includes("/")) {
       presenceData.startTimestamp = browsingStamp;
-      presenceData.details = "Viewing 创作中心";
+      presenceData.details = "Their dashboard";
```
Comment (Bas950) [inline websites/B/bilibili/presence.ts:349]: `Viewing their dashboard`

File: websites/B/bilibili/presence.ts [hunk 6]
```diff
@@ -843,60 +412,61 @@ presence.on("UpdateData", async () => {
...
       );
       presenceData.startTimestamp = browsingStamp;
-      presenceData.details = "Browsing for live category";
-      presenceData.state = "└ " + title.innerText;
+      presenceData.details = "Viewing live channel";
+      presenceData.state = title.innerText;
     } else {
       presenceData.startTimestamp = browsingStamp;
-      presenceData.details = "Browsing for live channels";
+      presenceData.details = "Viewing live channel";
```
Comment (Bas950) [inline websites/B/bilibili/presence.ts:429]: channels* you forgot an s at the end

File: websites/B/bilibili/presence.ts [hunk 7]
```diff
@@ -843,60 +412,61 @@ presence.on("UpdateData", async () => {
...
         " - 搜索结果 - 哔哩哔哩弹幕视频网 - ( ゜- ゜)つロ  乾杯~  - bilibili",
         ""
       );
+      presenceData.smallImageKey = "search";
     }
   } else if (document.location.hostname == "message.bilibili.com") {
     if (document.URL.includes("/#/reply")) {
       presenceData.startTimestamp = browsingStamp;
-      presenceData.details = "Viewing message";
+      presenceData.details = "Their message";
```
Comment (Bas950) [inline websites/B/bilibili/presence.ts:445]: their message sounds kinda strange aswell

File: websites/B/bilibili/presence.ts (hunk 1 above)
Comment (Kotoki1337) [reply websites/B/bilibili/presence.ts:130]: > `Viewing their history` might be better
> Viewing history or Their history is both not really good.

i saw that youtube uses "their history" so i copied it

File: websites/B/bilibili/presence.ts (hunk 3 above)
Comment (Kotoki1337) [reply websites/B/bilibili/presence.ts:690]: > Why was this all removed?

these things are very deep, and no one will go into these places. i am thinking of other more concise ways to implement them. some places that no one visits and take up too much code block. i think it is a bit too messy.

Comment (Bas950) [reply websites/B/bilibili/presence.ts:690]: If it works then there is no reason to remove it, even if close to noone goes there.

Comment (Kotoki1337) [reply websites/B/bilibili/presence.ts:690]: > If it works then there is no reason to remove it, even if close to noone goes there.

i implemented it in a more concise way

**Pull request merged**